python/MooseDocs/test/__init__.py

```diff
@@ -53,6 +57,22 @@ def __setup(self, reader=None, renderer=None, extensions=None, executioner=None)
         self.__translator.init()
         self.__translator.execute() # This is required to setup the meta data
 
+    def setUp(self):
+        # Wrap mooseutils.runExe so that we don't call the application
+        # with the same arguments more than once. The first time with
+        # a given set of arguments, it'll run the app. Every time after
+        # that, it'll use the cache
+        def run_exe_cached(app_path, args):
+            key = app_path + ''.join(args)
```

Comment:
Not 

```suggestion
            key = app_path + '' .join(args)
```

?

Reply:
What??

I think you mean:
```
key = app_path + ' '.join(args)
```
?

This isn't ever read by a human. It's just a squash of all of the things to make a hashable key based on the app and args

Reply:
I know, but space separation removes a potential ambiguity. Squashing arguments together would make different argument sets that have spaces in different spots equivalent. I cannot see a potential combo for only our options, but if you add in hit parameter changes, it could lead to obscure bugs that are super easy to avoid by adding that single space.

```
./moose-opt -i input.i xy=y xz=y
./moose-opt -i input.i xy=yx z=y
```

Reply:
This is ran for:

- /path/to/moose_test-opt --json --allow-test-objects
- /path/to/moose_test-opt --show-type

You're definitely right. But... I'm not particularly concerned and am a bit checked out after four python fixes in a day